docs/help/FAQ/Puppet.rst

```diff
@@ -5,6 +5,37 @@ Puppet-Related Issues
 
 .. contents:: :local:
 
+.. _faq-puppet-debug_certs:
+
+What do I do when puppet gets certificate verification errors?
+--------------------------------------------------------------
+
+If you are experiencing an error when running puppet such as ``certificate
+verify failed`` then there are a few things that you can try in an attempt to
+troubleshoot the issue.
+
+#. Make sure that your system clocks within one hour of each other.
+#. Ensure that the forward and reverse lookup for the FQDN of your systems is
+   correct and matches the hostnames listed in the output of
+   ``openssl x509 -text -noout -in $(puppet config print hostcert) | less``
+
+   HINT: Look at the `Subject` and `X509v3 Subject Alternative Name` sections.
+
+#. Check that the connection from the client system to the server can
+   successfully connect:
+
+   .. code-block:: bash
+
+      openssl s_client -host $(puppet config print server) \
+      -port $(puppet config print masterport) \
+      -cert $(puppet config print hostcert) \
+      -key $(puppet config print hostprivkey) \
+      -CAfile $(puppet config print cacert)
```

Comment:
When I tried this on the puppet master, it worked.  When I tried it on another host it did not. openssl could not `read /etc/puppetlabs/puppet/ssl/ca/ca_crt.pem` in the `X505_load_cert_crl_file` routine.

Reply:
worked for me!

Reply:
Doesn't work for me.  the puppet ca cert seems to be at //etc/puppetlabs/puppet/ssl/certs/ca.pem on our systems.  Don't know if it is our setup or SIMP or puppet.  Need to look at pupmod to see what it does.

Reply:
Found it, need to use localcacert.  I will change it.  cacert is the location on the server.